addons/stock/report/report_stock_rule.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class ReportStockRule(models.AbstractModel):
+    _name = 'report.stock.report_stock_rule'
+
+    @api.model
+    def _get_report_values(self, docids, data=None):
+        product = self.env['product.product'].browse(data['product_id'])
+        warehouses = self.env['stock.warehouse'].browse(data['warehouse_ids'])
+
+        routes = self._get_routes(data)
+
+        # Some routes don't have a warehouse_id but contain rules of diferent warehouses
```

Comment:
Some routes don't have a warehouse_id but contain rules of different warehouses, we filter here the ones we want to display and build for each one a dict containing the rule and their source and destination location.